app/uk/gov/hmrc/cgtpropertydisposalsfrontend/connectors/onboarding/IvConnector.scala

```diff
@@ -40,22 +40,21 @@ class IvConnectorImpl @Inject() (
   servicesConfig: ServicesConfig
 )(implicit
   ec: ExecutionContext
-) extends IvConnector {
+) extends IvConnector with Logging {
 
   val baseUrl: String = servicesConfig.baseUrl("iv")
 
-  def url(journeyId: UUID): String =
-    s"$baseUrl/mdtp/journey/journeyId/${journeyId.toString}"
+  def url(journeyId: UUID): String = s"$baseUrl/mdtp/journey/journeyId/${journeyId.toString}"
 
   override def getFailedJourneyStatus(
     journeyId: UUID
-  )(implicit hc: HeaderCarrier): EitherT[Future, Error, HttpResponse] =
-    EitherT[Future, Error, HttpResponse](
-      http
-        .GET[HttpResponse](url(journeyId))
-        .map(Right(_))
-        .recover { case e =>
-          Left(Error(e))
-        }
-    )
+  )(implicit hc: HeaderCarrier): EitherT[Future, Error, IvStatusResponse] =
+    http
+      .GET[Either[UpstreamErrorResponse, IvStatusResponse]](url(journeyId))
+      .map(_.left.map { error =>
+        logger.error(s"GET to ${url(journeyId)} failed", error)
+        Error(s"Response to address lookup came back with status ${error.statusCode}")
```

Comment:
Can we use 'UpstreamErrorResponse' instead of 'Response'

Reply:
We can, but it doesn't make it any more accurate. The UpstreamErrorResponse is an object representing a Response. UpstreamErrorResponses aren't sent by downstream, Responses are.